databricks/provider.go

```diff
@@ -197,6 +218,41 @@ func providerConfigureAzureClient(d *schema.ResourceData, providerVersion string
 	return &dbClient, nil
 }
 
+// tryDatabricksCliConfigFile sets Host and Token from ~/.databrickscfg file if it exists
+func tryDatabricksCliConfigFile(d *schema.ResourceData, config *service.DBApiClientConfig) error {
+	configFile, err := homedir.Expand(d.Get("config_file").(string))
+	if err != nil {
+		return err
+	}
+	cfg, err := ini.Load(configFile)
+	if err != nil {
+		return fmt.Errorf("Authentication is not configured for provider. Please configure it\n"+
+			"through one of the following options:\n"+
+			"1. DATABRICKS_HOST + DATABRICKS_TOKEN environment variables.\n"+
+			"2. host + token provider argumeents.\n"+
+			"3. Run `databricks configure --token` that will create %s file.\n\n"+
+			"Please check https://bit.ly/2XCtuZU for details", configFile)
```

Comment:
I'm out of terraform context, but I would not understand that I need to create an `ini` file with host and token in it by reading this error message. Why it's not just printed to console instead of passing as an error, by the way? In other places we return internal errors, do we print those directly to the client?
Why is it an `ini` and not `json` or `yaml` by the way?

Reply:
Terraform prints errors in it's own way, so i just rely on standard mechanism of it's framework.

Regarding INI files - 🤷, https://docs.databricks.com/dev-tools/cli/index.html#set-up-authentication, https://github.com/databricks/databricks-cli/blob/master/databricks_cli/configure/provider.py - most likely because https://docs.python.org/3/library/configparser.html is an INI parser